kotlinlib/src/mill/kotlinlib/contrib/ktfmt/KtfmtModule.scala

```diff
@@ -0,0 +1,139 @@
+package mill
+package kotlinlib.contrib.ktfmt
+
+import mainargs.Leftover
+import mill.api.{Loose, PathRef}
+import mill.define.{Discover, ExternalModule}
+import mill.kotlinlib.{DepSyntax, KotlinModule}
+import mill.main.Tasks
+import mill.util.Jvm
+
+trait KtfmtBaseModule extends KotlinModule {
+
+  /**
+   * Classpath for running Ktfmt.
+   */
+  def ktfmtClasspath: T[Loose.Agg[PathRef]] = T {
+    defaultResolver().resolveDeps(
+      Agg(ivy"com.facebook:ktfmt:${ktfmtVersion()}")
+    )
+  }
+
+  /**
+   * Ktfmt version.
+   */
+  def ktfmtVersion: T[String] = T {
+    "0.52"
+  }
+
+  /**
+   * Additional arguments for Ktfmt. Check
+   * [[https://github.com/facebook/ktfmt/blob/main/core/src/main/java/com/facebook/ktfmt/cli/ParsedArgs.kt#L51 available options]].
+   */
+  def ktfmtOptions: T[Seq[String]] = T {
+    Seq.empty[String]
+  }
+}
+
+/**
+ * Performs formatting checks on Kotlin source files using [[https://github.com/facebook/ktfmt Ktfmt]].
+ */
+trait KtfmtModule extends KtfmtBaseModule {
+
+  /**
+   * Runs [[https://github.com/facebook/ktfmt Ktfmt]]
+   *
+   * @param ktfmtArgs arguments for the [[https://github.com/facebook/ktfmt Ktfmt]].
+   * @param sources list of sources to run  the tool on. If not provided, default module sources will be taken.
+   */
+  def ktfmt(
+      @mainargs.arg ktfmtArgs: KtfmtArgs,
+      @mainargs.arg(positional = true) sources: Leftover[String]
+  ): Command[Unit] = Task.Command {
+    val _sources = if (sources.value.isEmpty) {
+      this.sources()
+    } else {
+      sources.value.iterator.map(rel => PathRef(millSourcePath / os.RelPath(rel)))
+    }
+    KtfmtModule.ktfmtAction(
+      ktfmtArgs.style,
+      ktfmtArgs.format,
+      ktfmtArgs.removeUnusedImports,
+      _sources,
+      ktfmtClasspath(),
+      ktfmtOptions()
+    )
+  }
+}
+
+object KtfmtModule extends ExternalModule with KtfmtBaseModule with TaskModule {
+
+  def kotlinVersion = "1.9.24"
```

Comment:
I don't like this hardcoded version here. We should source it from some generated `BuildInfo`.